vllm/v1/sample/ops/topk_topp_sampler.py

```diff
@@ -199,7 +224,7 @@ def apply_top_k_only(
     max_top_k = k.max()
     # topk.values tensor has shape [batch_size, max_top_k].
     # Convert top k to 0-based index in range [0, max_top_k).
-    k_index = k.sub_(1).unsqueeze(1)
+    k_index = k.sub_(1).unsqueeze(1).expand(logits.shape[0], 1)
```

Comment:
@hyeygit is this because of a TPU torch broadcasting limitation?

Reply:
Yes I think so. Without the explicit expand this fails on XLA.

Since TPU no longer uses this code path, I can revert this change if needed.